examples/notebooks/sft_trl_lora_qlora.ipynb

```diff
@@ -1,1107 +1,1146 @@
+        "print(generated_text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ZLzGbCxskqhm"
+      },
+      "source": [
+        "The model now generates its reasoning trace in German!"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "43eDP89Giw_G"
+      },
+      "source": [
+        "## Inference and Serving with vLLM\n",
+        "\n",
+        "You can use Transformer models with **vLLM** to serve them in real-world applications. Learn more [here](https://blog.vllm.ai/2025/04/11/transformers-backend.html)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "_lTPhsXgTnvr"
+      },
+      "outputs": [],
+      "source": [
+        "!pip install -qU vllm"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "OrhC_Ao4iw_G"
+      },
+      "source": [
+        "### Push Merged Model (for LoRA or QLoRA Training)\n",
+        "\n",
+        "To serve the model via **vLLM**, the repository must contain the merged model (base model + LoRA adapter). Therefore, you need to upload it first."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "frKz1y1RTnvs"
+      },
+      "outputs": [],
+      "source": [
+        "model_merged = fine_tuned_model.merge_and_unload()\n",
+        "\n",
+        "save_dir = f\"{output_dir}-merged\"\n",
+        "\n",
+        "model_merged.save_pretrained(save_dir)\n",
+        "tokenizer.save_pretrained(save_dir)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "q-PryFrVTnvs"
+      },
+      "outputs": [],
+      "source": [
+        "model_merged.push_to_hub(f\"sergiopaniego/{output_dir}-merged\") # Replace with your HF username or organization\n",
+        "tokenizer.push_to_hub(f\"sergiopaniego/{output_dir}-merged\") # Replace with your HF username or organization"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yHMEf-FIiw_G"
+      },
+      "source": [
+        "### Performing Inference with vLLM\n",
+        "\n",
+        "Use **vLLM** to run your model and generate text efficiently in real-time. This allows you to test and deploy your fine-tuned models with low latency and high throughput."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "O97pSHceTnvs"
+      },
+      "outputs": [],
+      "source": [
+        "from vllm import LLM, SamplingParams\n",
+        "from transformers import AutoTokenizer\n",
+        "import torch\n",
+        "\n",
+        "llm = LLM(\n",
+        "    model=f\"sergiopaniego/{output_dir}-merged\", # Replace with your HF username or organization\n",
```

Comment:
no need to do it right now but in the future it could be nice to centralize every artifact in hf.co/trl-lib